### ISSUE TITLE
Remove python3-distutils-extra as a runtime dependency

### ISSUE THREAD

**Issue by gornostal** [Ulauncher/Ulauncher#500]
`python3-distutils-extra` depends on a lot of stuff but Ulauncher only needs `mkpath` and `StrictVersion`

It's better to remove this dependency and implement `mkpath` and `StrictVersion` in Ulauncher

**Comment by gapplef** [comment]
I'm not familiar with the detail code, but some questions:

1. Isn't `python3-distutils` enough to provide these functions, why is `python3-distutils-extra` needed?

2. For [`mkpath`](https://docs.python.org/3.8/distutils/apiref.html#distutils.dir_util.mkpath), seems it's just a function for creating directories, is it possible to use [`pathlib.Path.mkdir` or `os.makedirs`](https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory) instead?

3. For [`StrictVersion`](https://docs.python.org/3.8/distutils/apiref.html#module-distutils.version), no description found in official docs, but for version comparing seems [`packaging`](https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python/21065570) is more recommend. Also I found [this](https://github.com/peopledoc/septentrion/pull/39/commits/b11cf741a9d7d09bdf226ee05ddafad7b87c941a) for reference.

**Comment by gornostal** [comment]
You are right, `os.makedirs` should be sufficient enough.

> or version comparing seems packaging is more recommend. Also I found [this](https://github.com/peopledoc/septentrion/pull/39) for reference.

Yep. Could be a good solution

**Comment by friday** [comment]
The [only place](https://github.com/Ulauncher/Ulauncher/blob/c23da1aa04d60fd2a710e5f6c2dd30772980c44e/ulauncher/utils/version_cmp.py) `StrictVersion` is used, both strings it compares are concatenated by Ulauncher, so it's easy to replace it with an uglier long condition:

```py
def gtk_version_is_gte(major: int, minor: int, micro: int) -> bool:
    gtk_version = {'major': Gtk.get_major_version(), 'minor': Gtk.get_minor_version(), 'micro': Gtk.get_micro_version()}
    return gtk_version['major'] > major or (gtk_version['major'] == major and gtk_version['minor'] > minor or (gtk_version['minor'] == minor and gtk_version['micro'] >= micro))
```

I don't have the Ulauncher dev environment or I would make that a PR.

**Comment by friday** [comment]
Fixed by #554
